src/data/getCategoriesByMovies.js

```diff
@@ -0,0 +1,26 @@
+import { fetchData } from "./fetchData.js";
+import { getGenres } from "./getGenres.js";
+
+export const getCategoriesByMovies = async (movies) => {
+  let genres;
+  if (movies) {
+    genres = await getGenres(movies);
+  }
+
+  //const categories = await getCategoriesByMovies(genres);
+
+  const response = await fetchData(
+    "https://api.themoviedb.org/3/genre/movie/list"
+  );
+
+  if (response) {
+    const categories = response.genres.filter((category) => {
+      return genres.includes(category.id);
+    });
+    return categories;
```

Comment:
This can be coded more concisely. Also, you introduced the name `category` while the property on the response object is called `genre`. Better to stick to the actual name.


```suggestion
    return response.genres.filter((genre) => genre.id);
```